### PR TITLE
bpo-47120: make JUMP_NO_INTERRUPT relative

### PR DESCRIPTION
**Pull request by iritkatriel**
https://bugs.python.org/issue47120

### CODE REVIEW

File: Doc/whatsnew/3.11.rst
Comment (iritkatriel) [inline Doc/whatsnew/3.11.rst]:
```suggestion
* Added :opcode:`JUMP_BACKWARD_NO_INTERRUPT`, which is used in certain loops where it is undesirable to handle interrupts.
```

(this is a new opcode, and it was never added to whatsnew).

**Review by brandtbucher — APPROVED** [review]
Nice. These are always tiny jumps too.

**Comment by markshannon** [comment]
This would be more robust, if you handled it the same way as `JUMP_FORWARD`/`JUMP_BACKWARD`.
Add a `JUMP_NO_INTERRUPT` virtual instruction, and let the assembly emit the correct opcode.
`JUMP_NO_INTERRUPT forward` is just `JUMP_FORWARD`.

**Comment by iritkatriel** [comment]
> This would be more robust, if you handled it the same way as `JUMP_FORWARD`/`JUMP_BACKWARD`. Add a `JUMP_NO_INTERRUPT` virtual instruction, and let the assembler emit the correct opcode. `JUMP_NO_INTERRUPT forward` is just `JUMP_FORWARD`.

Agreed. I'll update this PR after we've merged https://github.com/python/cpython/pull/32200 because it would clash otherwise.

**Review by raghavthind2005 — APPROVED** [review]

**Comment by markshannon** [comment]
Looks good.
Feel free to share the new magic number with https://github.com/python/cpython/pull/32309. One magic number per day should be enough.

**Comment by markshannon** [comment]
@iritkatriel I've leave it up to you to merge this and https://github.com/python/cpython/pull/32309 in whichever order you prefer

**Pull request merged**
